concent_api/core/views.py

```diff
@@ -1241,6 +1209,712 @@ def store_message_and_message_status(
+        )
+
+
+def update_states_of_subtasks(
+    subtask_list:       list,
+):
+    for subtask in subtask_list:
+        if subtask.state == Subtask.SubtaskState.FORCING_REPORT.name:  # pylint: disable=no-member
+            update_subtask_state(
+                subtask                 = subtask,
+                state                   = Subtask.SubtaskState.REPORTED.name,  # pylint: disable=no-member
+            )
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.AckReportComputedTask,
+                client_public_key   = subtask.provider.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.Receive,
+                subtask             = subtask,
+            )
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.VerdictReportComputedTask,
+                client_public_key   = subtask.requestor.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.ReceiveOutOfBand,
+                subtask             = subtask,
+            )
+        elif subtask.state == Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name:  # pylint: disable=no-member
+            update_subtask_state(
+                subtask                 = subtask,
+                state                   = Subtask.SubtaskState.FAILED.name,  # pylint: disable=no-member
+            )
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.ForceGetTaskResultFailed,
+                client_public_key   = subtask.requestor.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.Receive,
+                subtask             = subtask,
+            )
+        elif subtask.state == Subtask.SubtaskState.FORCING_ACCEPTANCE.name:  # pylint: disable=no-member
+            update_subtask_state(
+                subtask                 = subtask,
+                state                   = Subtask.SubtaskState.ACCEPTED.name,  # pylint: disable=no-member
+            )
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.ForceSubtaskResultsSettled,
+                client_public_key   = subtask.provider.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.Receive,
+                subtask             = subtask,
+            )
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.ForceSubtaskResultsSettled,
+                client_public_key   = subtask.requestor.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.ReceiveOutOfBand,
+                subtask             = subtask,
+            )
+        # TODO when last use case will be implemented
+        # elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION:
```

Comment:
Add an assert that checks that all active states except `ADDITIONAL_VERIFICATION` are handled here.

Also add an assert that fails if the state is `ADDITIONAL_VERIFICATION`. When we add verification it will fail reminding us to fix this place (it's much better than just a comment that we can easily forget).

Reply:
Oh, so you're using "thumbs up" as marker that a particular change mentioned in a comment was done? That's a nice convention. I like it.

I thought you're using it just as "agree and I have nothing more to add". The icon suggests that. Maybe we could use a different icon to make the distiction clearer. For example the one I'm putting on this comment?